src/OpenTK.Windowing.Desktop/NativeWindow.cs

```diff
@@ -953,29 +953,25 @@ private unsafe void ProcessInputEvents()
 
             for (var i = 0; i < _joystickStates.Length; i++)
             {
-                var joy = _joystickStates[i];
-                if (joy == default)
+                if (_joystickStates[i] == default)
```

Comment:
This is not a "well" defined equality check, it looks like this is a null check but it's actually a memberwise struct comparison. I think `JoystickState` should be a class (and maybe change name idk).

Reply:
JoystickState is a class, and _joystickStates[i] is IJoystickState, so in this circumstance it will be a null check

Reply:
(The code has changed to null, you might need to refresh: at the top of this it says "Outdated")